### PR TITLE
YLP-2528 The "Cartridge changes" widget doesn't display the info on the 14 days period

### PR DESCRIPTION
**Pull request by TimCanu**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

File: packages/tideline/plugins/blip/basics/chartbasicsfactory.js [hunk 1]
```diff
@@ -55,13 +55,23 @@ class BasicsChartNoSize extends React.Component {
     }
 
     const basicsData = _.cloneDeep(tidelineData.basicsData)
+
+    // We only want the last 15 days for the dashboard
+    const now = new Date()
+    const twoWeeksAgo = new Date()
+    twoWeeksAgo.setDate(now.getDate() - 15)
```
Comment (celia-rouis) [inline packages/tideline/plugins/blip/basics/chartbasicsfactory.js:62]: **🎯 suggestion (non-blocking):** ‏It might be nice to reuse `DEFAULT_DASHBOARD_TIME_RANGE_DAYS` from `patient-data.util.ts`, in order to have everything working if we ever change the dashboard time range again.
Also, in that spirit, `twoWeeksAgo` could be renamed `dashboardStartDate`

File: packages/tideline/plugins/blip/basics/components/CalendarContainer.js [hunk 2]
```diff
@@ -129,6 +129,9 @@ class CalendarContainer extends React.Component {
     // Could be subject to change so I thought this was preferred over
     // hard-coding a solution that assumes Monday is the first day
     // of the week.
+    if (this.props.days.length === 0) {
```
Comment (celia-rouis) [inline packages/tideline/plugins/blip/basics/components/CalendarContainer.js:132]: **❔ question:** ‏Why is this necessary? (also, the comment belongs to the code below what you added)

**Review by celia-rouis — APPROVED** [review]

File: packages/tideline/plugins/blip/basics/components/CalendarContainer.js (hunk 2 above)
Comment (TimCanu) [reply packages/tideline/plugins/blip/basics/components/CalendarContainer.js:132]: It seems that some patient could get an application crash because of the line:
```
var firstDay = moment.utc(this.props.days[0].date).day()
```
`this.props.days` can now be an empty array, meaning that `this.props.days[0]` can be undefined and `this.props.days[0].date` throws an NPE
